src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_ETTIN:
+                {
+                    if (Creature* ettin = me->FindNearestCreature(NPC_FOREST_ETTIN, 40.0f))
+                    {
+                        me->SetFacingToObject(ettin);
+
+                        _events.ScheduleEvent(EVENT_CHECK_ETTIN + 1, 1s);
+                    }
+                    else
+                        me->DespawnOrUnsummon(2s);
+                    break;
+                }
+
+                case EVENT_CHECK_ETTIN + 1:
+                    if (Creature* ettin = me->FindNearestCreature(NPC_FOREST_ETTIN, 40.0f))
+                        me->GetMotionMaster()->MoveFollow(ettin, 2.0f, float(M_PI * 2));
+                    _events.ScheduleEvent(EVENT_CHECK_ETTIN + 2, 1s);
+                    break;
+
+                case EVENT_CHECK_ETTIN + 2:
+                {
+                    if (Creature* ettin = me->FindNearestCreature(NPC_FOREST_ETTIN, 15.0f))
+                    {
+                        ettin->SetFacingToObject(me);
+
+                        ettin->PauseMovement();
+
+                        _events.ScheduleEvent(EVENT_CHECK_ETTIN + 3, 2s);
+                    }
+                    else
+                        _events.ScheduleEvent(EVENT_CHECK_ETTIN + 2, 1s);
+                    break;
+                }
+
+                case EVENT_CHECK_ETTIN + 3:
+                {
+                    if (Creature* ettin = me->FindNearestCreature(NPC_FOREST_ETTIN, 5.0f))
+                    {
+                        if (ettin->IsAIEnabled())
+                            ettin->AI()->DoAction(ACTION_GRAB_CHICKEN);
+                    }
+                    break;
+                }
+
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+enum ForestEttin
+{
+    NPC_MUTANT_BUSH_CHICKEN                     = 44935,
+
+    SPELL_HAULING_TIMBER                        = 88361,
+    SPELL_REVERSE_CAST_RIDE_VEHICLE_ETTIN       = 83904,
+    SPELL_ETTIN_MOUTH                           = 83907,
+    SPELL_BUSH_EXPLOSION                        = 83903,
+    SPELL_BONK                                  = 80146,
+    SPELL_LOG_SMASH                             = 88421,
+
+    EVENT_LOG_SMASH                             = 1,
+    EVENT_BONK                                  = 2,
+    EVENT_CHICKEN_EXPLODE                       = 3,
+
+    SEAT_ETTIN_ARM                              = 0,
+    SEAT_ETTIN_MOUTH                            = 1
+};
+
+// 44367 - Forest Ettin
+struct npc_silverpine_forest_ettin : public ScriptedAI
+{
+    npc_silverpine_forest_ettin(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        me->setActive(true);
```

Comment:
setActive is very very very evil, we shouldnt use that here as it keeps the grid loaded

Reply:
I can remove it but since visibilityDistance is 3 it will be seen stuck in one place from hundred yards away until the player gets near to it to load the grid.

Reply:
Removed, feedback applied.